pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -206,7 +214,24 @@ func (p *PollerWorker) Poll(execAgentCommands agentCommandsExecutor, closeChan c
 		case <-ticker.C:
 			execAgentCommands(p.AgentCommands)
 		}
+		if pollInterval < pollStableInterval {
+			pollInterval = incrementPollInterval(pollInterval, pollStableInterval)
+			ticker = replaceTicker(ticker, pollInterval)
+		}
+	}
+}
+
+func replaceTicker(ticker *time.Ticker, interval time.Duration) *time.Ticker {
+	ticker.Stop()
+	return time.NewTicker(time.Second * interval)
+}
+
+func incrementPollInterval(interval time.Duration, maxInterval time.Duration) time.Duration {
+	interval *= 2
+	if interval > maxInterval {
```

Comment:
Isn't this instruction redundant ? You only call it `if pollInterval < pollStableInterval` , and in the scope of this function:

- interval => pollInterval
- pollStableInterval => maxInterval

Unless I'm somehow reading this wrong, you could just have it like:

```golang 
func incrementPollInterval(interval time.Duration) time.Duration {
    return 2 * interval
}
```

and let the caller control the flow of time (it already does it).

Reply:
Oh I see it; you want to forcefully coerce the max val. 

OK, no need to reply, you can resolve this.